main.c

```diff
@@ -147,7 +182,7 @@ IOTHUBMESSAGE_DISPOSITION_RESULT receiveMessageCallback(IOTHUB_MESSAGE_HANDLE me
     strncpy(temp, buffer, size);
     temp[size] = '\0';
 
-    printf("Receiving message: %s\r\n", temp);
+    (void)printf("Receiving message: %s\r\n", temp);
```

Comment:
I think the LogInfo can be closed by some flag, but in the C2D message, it should always print out.